Simplify abus_current_url

philschoefer opened the pull request:
I'm having trouble with this function on pantheon.io. The reason is that the $_SERVER[ 'SERVER_PORT' ] variable is actually 9172 or something and the site still runs on 80 though. I'm not sure exactly why that is the case but I assume it has to do with the fact that everything is running in docker containers which each have their own port.

I've ran some experiments with the code above, even on a local MAMP running on :8000 and it's working just fine with this code update. 

This code update would also honour whatever setting was done within WordPress in terms of which protocol should be used etc.

Am I missing anything?

philschoefer committed:
Simplify abus_current_url

I'm having trouble with this function on pantheon.io. The reason is that the $_SERVER[ 'SERVER_PORT' ] variable is actually 9172 or something and the site still runs on 80 though. I'm not sure exactly why that is the case but I assume it has to do with the fact that everything is running in docker containers which each have their own port.

I've ran some experiments with the code above, even on a local MAMP running on :8000 and it's working just fine with this code update. 

This code update would also honour whatever setting was done within WordPress in terms of which protocol should be used etc.

Am I missing anything?

wpmark commented:
I was actually thinking of removing this function altogether and then every time the user switch takes place redirect the user to the dashboard in the admin.

The reason for this is that if you are on a page where the user you are switching to does not have permissions, for example you are switching to a user with subscriber permissions and you are on a settings page, you get a `wp_die()` error which is not great.

philschoefer commented:
HI Mark,

that makes sense. Keep in mind that this plugin is also used by people who log in as customers in Woocommerce or other user roles that don’t have admin dashboard access altogether. In those cases it would need to redirect to “My Account”. 

Actually it seems like WordPress does that by default so what you’re suggesting should work just fine.

I’m for adding a hook there that allows someone to overwrite what you’re being redirected to. I can draft something up within the next couple of days.

---

PHILIPP SCHÖFFER 
Webdesign and Development

PERSPEKTIVE WEB + DESIGN // Max-Reger-Str. 6 · 69121 Heidelberg, Germany · +49 6221 73 568 73
ps@perspektive-web.de · www.perspektive-web.de

> On Apr 13, 2017, at 4:12 AM, Mark Wilkinson <notifications@github.com> wrote:
> 
> I was actually thinking of removing this function altogether and then every time the user switch takes place redirect the user to the dashboard in the admin.
> 
> The reason for this is that if you are on a page where the user you are switching to does not have permissions, for example you are switching to a user with subscriber permissions and you are on a settings page, you get a wp_die() error which is not great.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/wpmark/admin-bar-user-switching/pull/14#issuecomment-293823031>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AEuv2XocBGZzN-XoJcoWxqG1Y7TWn481ks5rvdkEgaJpZM4M76Rs>.
>

Commit:
Replaced get_current_url function function with get redirect function and added filter to be able to overwrite redirect.

Commit:
added filter for redirect

Commit:
Merge branch 'master' into patch-1

# Conflicts:
#	admin-bar-user-switching.php

philschoefer commented:
Updated my code. I replaced the abus_get_current_url function with abus_get_redirect_url and added a filter.

There are also some style adaptations. I added CSS so the dropdown reflects the admins chosen color schema and added a spinner gif to the loading time from the woocommerce assets directory.

philschoefer committed:
Merge branch 'master' into patch-1

philschoefer commented:
@wpmark it's been a while ;). I'm just checking in if you're still interested in this update.

wpmark commented:
Sorry about this, it has not been priority number 1! I will try to action this over the coming weeks.

philschoefer commented:
No worries, not in a rush here either.

wpmark commented:
There is an issue here with using an asset from the WooCommerce plugin as most won't have that plugin active.

philschoefer commented:
Thanks for pointing that out. I'll make sure to remove that 👍

philschoefer commented:
Going to close this. It seems like it's not an issue that a lot of people deal with and I don't have the capacity to follow this through right now.
